File: parishmis/parish_management_system/report/parishioners_per_group/parishioners_per_group.py
import frappe


def execute(filters=None):
    filters = filters or {}
    group_type = filters.get("group_type")

    where = ""
    params = {}
    if group_type:
        where = " WHERE g.group_type = %(group_type)s"
        params["group_type"] = group_type

    data = frappe.db.sql(
        f"""
        SELECT
            g.group_type AS group_type,
            COUNT(m.name) AS total_members
        FROM `tabMembership` m
        INNER JOIN `tabGroup` g ON g.name = m.group
        {where}
        GROUP BY g.group_type
        ORDER BY g.group_type
        """,
        params,
        as_dict=True,
    )

    columns = [
        {"label": "Group", "fieldname": "group_type", "fieldtype": "Data", "width": 200},
        {"label": "Total Members", "fieldname": "total_members", "fieldtype": "Int", "width": 140},
    ]
    return columns, data

File: parishmis/parish_management_system/report/parishioners_per_group/parishioners_per_group.js
frappe.query_reports["Parishioners per Group"] = {
	"filters": [
		{
			"fieldname": "group_type",
			"label": __("Group Type"),
			"fieldtype": "Select",
			"options": "\nSCC\nCWA\nCMA\nLegion\nOther",
			"on_change": function() {
				frappe.query_report.set_filter_value('group', "");
			}
		},
		{
			"fieldname": "group",
			"label": __("Group"),
			"fieldtype": "Link",
			"options": "Group",
			"get_query": function() {
				const group_type = frappe.query_report.get_filter_value('group_type');
				if (group_type) {
					return {
						filters: {
							"group_type": group_type
						}
					};
				}
			}
		},
		{
			"fieldname": "parish",
			"label": __("Parish"),
			"fieldtype": "Link",
			"options": "Parish"
		},
		{
			"fieldname": "status",
			"label": __("Membership Status"),
			"fieldtype": "Select",
			"options": "\nActive\nInactive"
		}
	]
};
